lib/_http_client.js

```diff
@@ -169,6 +172,15 @@ function ClientRequest(input, options, cb) {
   if (options.timeout !== undefined)
     this.timeout = getTimerDuration(options.timeout, 'timeout');
 
+  const signal = options.signal;
+  if (signal) {
+    validateAbortSignal(signal, 'options.signal');
+    const listener = (e) => this.destroy();
```

Comment:
I would call `.destroy()` with an error.

Reply:
I... think you're right. If cancellation is abnormal elsewhere it should likely be abnormal termination here and abort with an AbortError with a .code

Reply:
I... think I changed my mind again. Lol, I think calling `.destroy` without an error is probably right and the error the user is getting should probably be `ECONNRESET` and not an `AbortError` anyway.

I'd rather users get consistent errors on request aborts than always getting `AbortError`s consistent errors when using AbortController.

I'm really not sure here tbh.

Reply:
cc @ronag

Reply:
I don't think it will ECONNRESET if aborted before a socket is assigned.

Reply:
@ronag it does (there is a test for it)

Reply:
cc @jasnell